generator/templates/wire/WireClient.cpp

```diff
@@ -517,18 +561,17 @@ namespace nxt { namespace wire {
                     }
 
                     //* The requests can have been deleted via an Unmap so this isn't an error.
-                    auto requestIt = buffer->readRequests.find(cmd->requestSerial);
-                    if (requestIt == buffer->readRequests.end()) {
+                    auto requestIt = buffer->requests.find(cmd->requestSerial);
+                    if (requestIt == buffer->requests.end() || requestIt->second.isWrite) {
                         return true;
                     }
 
                     auto request = requestIt->second;
-                    // Delete the request before calling the callback otherwise the callback could be fired a second time if for example buffer.Unmap() is called inside the callback.
-                    buffer->readRequests.erase(requestIt);
+                    //* Delete the request before calling the callback otherwise the callback could be fired a second time if for example if buffer.Unmap() is called inside the callback.
```

Comment:
Done, //* now has an official name 🗡 
It is a template comment so that the comments don't appear in the generated file. It isn't useful here but would be for things like generated headers we want to be semi-readable.